### DISCUSSION TITLE
Error 4

### DISCUSSION THREAD

**Discussion by dobos** [maca88/SmartBikeLights#77] [General]
Hi, I'm trying to use the configurator to set up my ION PRO / FLARE RT lights. I leave everything on default and enter the serial numbers only. I'm getting Error 4 in the widget and no buttons are displayed. I have an 830 and am using the Android app to upload the settings. Can a wrong serial number cause Error 4 or is it the syntax? I only see one of the serial numbers in the string below (replaced the last four digits with **** for privacy). The other number is 10 digits long, can that be the problem?

```
#4587520,196641:1,216354231:1##6291461,1409482753:0,8438****:1##7,6:Ion Pro RT:0:16777215!2,:-1,Off:0!1,High:1!1,Medium:2!1,Low:5!1,Day Flash:63!1,Night Flash:62#7,6:Flare RT:0:16777215!2,:-1,Off:0!1,Day Steady:1!1,Night Steady:5!1,Day Flash:7!1,All-Day Flash:8!1,Night Flash:63#0::#0:0#123!:123!#0#B3122#1#1#1#0
```

**Comment by maca88** [comment]
Hi, I tried the configuration in the Edge 830 simulator and it works fine. Are you sure that you are using Smart Bike Lights data fields and not Bike Lights Control widget? If yes, does it work if you create the configuration without setting the serial numbers?

**Comment by dobos** [comment]
Sorry for the delay, I finally had time to test this again. I indeed was using the widget so I've switched to the data screen. Now I can enter the string above, not error, but when I switch to the configuration that has the string entered, the data screen displays 'no network'. I can switch to another configuration which has no config string and the buttons are displayed properly. I tested it with a real 830.

**Comment by maca88** [comment]
If the buttons are displayed with no configuration and not with the configuration above, then the serial numbers are incorrect. If you have only two lights paired with Edge, then you can remove the serial numbers from the configuration and it should work. Try this configuration:

#4587520,196641::1##6291461,1409482753::1##7,6:Ion Pro RT:0:16777215!2,:-1,Off:0!1,High:1!1,Medium:2!1,Low:5!1,Day Flash:63!1,Night Flash:62#7,6:Flare RT:0:16777215!2,:-1,Off:0!1,Day Steady:1!1,Night Steady:5!1,Day Flash:7!1,All-Day Flash:8!1,Night Flash:63#0::#0:0#123!:123!#0#B3122#1#1#1#0

**Comment by dobos** [comment]
Turned out that the original one worked but I had to go back and forth between the configurations until suddenly the 'no network' that was displayed only when a config string was entered, disappeared. I couldn't narrow it down to anything but it is working now. Thanks for pointing out that I should use the data field instead of the widget.
